### PR TITLE
Service role

### PR DESCRIPTION
**Pull request by NikitaGandhi**
# Why this change is needed
> Describe why this change is needed, what issues it will fix and the benefits the change will add

Provision OpenID Connect (OIDC) identity provider, IAM roles for service-account and trusted policy to support fine-grained IAM role based access to pod level using service accounts. This change is made to test the functionality and address #122 
 
# Negative effects of this change
> Will making this change break or change an existing functionality? flag it here

No it is an optional addons.

### CODE REVIEW

File: addons/service_account_roles.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  #       EOF
+  #   }
+  # ]
+}
+
+resource "aws_iam_openid_connect_provider" "example" {
+  count     = (length(var.service_account_roles) > 0) ? 1 : 0
+  client_id_list  = ["sts.amazonaws.com"]
+  thumbprint_list = []
+  url             = "${data.aws_eks_cluster.eks.identity.0.oidc.0.issuer}"
```
Comment (tom-butler) [inline addons/service_account_roles.tf]: Where is this coming from? Should it be part of this add on?

Comment (NikitaGandhi) [reply addons/service_account_roles.tf]: Yes it creates Identity provider for OpenID Connect associate with cluster

Comment (NikitaGandhi) [reply addons/service_account_roles.tf]: it was defined in `main.tf`, so referencing it here
 
data "aws_eks_cluster" "eks" {
  name = var.cluster_name
}

**Comment by NikitaGandhi** [comment]
referencing data.aws_eks_cluster.eks from `main.tf`

**Pull request merged**